concent_api/core/views.py

```diff
@@ -181,28 +241,40 @@ def handle_send_ack_report_computed_task(client_message):
                 "Received AckReportComputedTask but RejectReportComputedTask "
                 "or another AckReportComputedTask for this task has already been submitted."
             )
+
+        assert force_task_to_compute.count() == 1, "More that one 'ForceReportComputedTask' found for this task_id."
+
         client_message.sig = None
         store_message_and_message_status(
             client_message.TYPE,
             client_message.task_to_compute.compute_task_def['task_id'],
             client_message.serialize(),
-            status = ReceiveStatus
+            provider_public_key  = force_task_to_compute.last().auth.provider_public_key_bytes,
+            requestor_public_key = client_public_key,
+            status               = ReceiveStatus,
         )
 
         return HttpResponse("", status = 202)
     else:
         raise Http400("Time to acknowledge this task is already over.")
 
 
-def handle_send_reject_report_computed_task(client_message):
-    current_time = int(datetime.datetime.now().timestamp())
+def handle_send_reject_report_computed_task(request, client_message):
+    current_time      = int(datetime.datetime.now().timestamp())
+    client_public_key = decode_client_public_key(request)
     validate_golem_message_reject(client_message.cannot_compute_task)
 
-    force_report_computed_task_from_database = Message.objects.filter(task_id = client_message.cannot_compute_task.task_to_compute.compute_task_def['task_id'], type = message.ForceReportComputedTask.TYPE)
+    force_report_computed_task_from_database = Message.objects.filter(
+        task_id                    = client_message.cannot_compute_task.task_to_compute.compute_task_def['task_id'],
+        type                       = message.ForceReportComputedTask.TYPE,
+        auth__requestor_public_key = request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY']
+    )
 
     if not force_report_computed_task_from_database.exists():
         raise Http400("'ForceReportComputedTask' for this task has not been initiated yet. Can't accept your 'RejectReportComputedTask'.")
```

Comment:
This message is now misleading. It can also mean that someone has sent a reject for someone else's task.

Reply:
:+1: Updated in both places.